annotated.go

```diff
@@ -585,11 +585,41 @@ func (at asAnnotation) apply(ann *annotated) error {
 	return nil
 }
 
+type fromAnnotation struct {
+	targets []interface{}
+}
+
+var _ Annotation = fromAnnotation{}
+
+// From is an Annotation that annotates the parameters for a function (i.e. a
+// constructor) that accepts interfaces. Very much the opposite of fx.As(...).
+func From(interfaces ...interface{}) Annotation {
+	return fromAnnotation{interfaces}
+}
+
+func (fr fromAnnotation) apply(ann *annotated) error {
+	if len(ann.From) > 0 {
+		return fmt.Errorf("fx.From does not support multiple annotations")
+	}
+	types := make([]reflect.Type, len(fr.targets))
+	for i, typ := range fr.targets {
+		t := reflect.TypeOf(typ)
+		if t.Kind() != reflect.Ptr || t.Elem().Kind() != reflect.Ptr && t.Elem().Kind() != reflect.Struct || t.Elem().Kind() == reflect.Ptr && t.Elem().Elem().Kind() != reflect.Struct {
```

Comment:
It's not immediately clear why we allow double pointer types to a struct.. What's the rationale behind allowing pointer to a struct type, and a pointer to a pointer to a struct type, but not pointer to a pointer to a struct type? 

I think it's more consistent if we _only_ allowed a pointer to a struct type.

Reply:
I agree with this so in this case the following test would fail (which is fine) due to `new` not being a ptr to a struct
`		{
			desc: "value type implementing interface",
			provide: fx.Provide(
				func() anotherStringer {
					return anotherStringer{
						"another stringer",
					}
				},
				fx.Annotate(
					func(myStringer myStringer) fmt.Stringer {
						return &fromStringer{
							name: myStringer.String(),
						}
					},
					fx.From(new(anotherStringer)),
				),
			),
			invoke: func(s fmt.Stringer) {
				assert.Equal(t, s.String(), "another stringer")
			},
		},`